### PR TITLE
Adaptions for the RESTful API

### PR DESCRIPTION
**Pull request by chaoxinhu**
## Summary
- Concisely describe the issue and your solution:
This CL mainly focus on the adaptions of the restful api change from weidentity-http-service (not open-source yet).
1. Adapation of new changes in RestAPI service (4 new interfaces)
2. Use compact Json with ordered Key output for Credential
3. Externalize Credential Param names and claim hash
4. TransactionUtils Unit Tests
5. Headers update

- Link to existing issue (type "#" to check): No
- Specify the tests done so far: Integration tests from http-service and SDK were done offline. UT added.

## Solutions (Optional)
Please describe the details of your solution if it might be too complicated for reviewers.

### CODE REVIEW

**Comment by chaoxinhu** [comment]
Hold this until next commit (code quality)

**Comment by chaoxinhu** [comment]
Open for reviews

File: src/main/java/com/webank/weid/util/CredentialUtils.java [hunk 1]
```diff
@@ -46,30 +47,75 @@
...
      */
-    public static String getCredentialFields(Credential arg, Map<String, Object> disclosures) {
-        String metaData = concatCredentialMetadata(arg);
-        if (StringUtils.isEmpty(metaData)) {
+    public static String getCredentialThumbprint(Credential credential,
+        Map<String, Object> disclosures) {
+        try {
+            Map<String, Object> credMap = JsonUtil.objToMap(credential);
+            String claimHash = getClaimHash(credential, disclosures);
+            credMap.replace(ParamKeyConstant.CLAIM, claimHash);
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/CredentialUtils.java]: replace有可能不成功的，建议用put。

File: src/main/java/com/webank/weid/util/DateUtils.java [hunk 2]
```diff
@@ -102,6 +102,20 @@ public static long converDateToTimeStamp(String time) throws ParseException {
         return ts;
     }
 
+    /**
+     * Conver utc date to time stamp.
+     *
+     * @param time the time
+     * @return the long
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/DateUtils.java]: 注释请补充一下，其他地方也一样。

File: src/main/java/com/webank/weid/util/TransactionUtils.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+            logger.error("Error processing transaction request: "
+                + ethSendTransaction.getError().getMessage());
+            return null;
+        }
+        String transactionHash = ethSendTransaction.getTransactionHash();
+
+        Optional<TransactionReceipt> receiptOptional =
+            getTransactionReceiptRequest(web3j, transactionHash);
+        int sumTime = 0;
+        for (int i = 0; i < WeIdConstant.POLL_TRANSACTION_ATTEMPTS; i++) {
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/TransactionUtils.java]: 重试这块，最好是有个默认的超时时间和重试次数的配置。

File: src/main/java/com/webank/weid/util/TransactionUtils.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    }
+
+    /**
+     * Convert the authority issuer org name into the static array.
+     *
+     * @param name authorityIssuer name
+     * @return the StaticArray
+     */
+    private static StaticArray<Bytes32> getParamName(String name) {
+        String[] nameArray = new String[WeIdConstant.AUTHORITY_ISSUER_ARRAY_LEGNTH];
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/TransactionUtils.java]: 我看代码里出现比较多的new String[WeIdConstant.AUTHORITY_ISSUER_ARRAY_LEGNTH]，这个长度有没有特殊的含义？

File: src/main/java/com/webank/weid/util/TransactionUtils.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+     *
+     * @return blocklimit in BigInt.
+     */
+    public static BigInteger getBlockLimit() {
+        try {
+            return BaseService.getWeb3j().ethBlockNumber().send().getBlockNumber()
+                .add(new BigInteger(String.valueOf(WeIdConstant.ADDITIVE_BLOCK_HEIGHT)));
+        } catch (Exception e) {
+            //Send a large enough block limit number
+            return new BigInteger("99999999999");
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/TransactionUtils.java]: 请用常量定义

File: src/main/java/com/webank/weid/service/impl/AuthorityIssuerServiceImpl.java [hunk 6]
```diff
@@ -167,6 +150,57 @@ private static void reloadContract(String privateKey) {
...
+     */
+    @Override
+    public ResponseData<String> registerAuthorityIssuer(String transactionHex) {
+        if (StringUtils.isEmpty(transactionHex)) {
+            logger.error("[registerAuthorityIssuer] hex value invalid.");
+            return new ResponseData<>(StringUtils.EMPTY, ErrorCode.ILLEGAL_INPUT);
+        }
+        try {
+            TransactionReceipt transactionReceipt = TransactionUtils
+                .sendTransaction(getWeb3j(), transactionHex);
```
[the remark this answers is not in the document]
Comment (chaoxinhu) [reply src/main/java/com/webank/weid/service/impl/AuthorityIssuerServiceImpl.java]: good

[the remark this answers is not in the document]
Comment (chaoxinhu) [reply src/main/java/com/webank/weid/service/impl/AuthorityIssuerServiceImpl.java]: 把检查空的抽出来了

File: src/main/java/com/webank/weid/util/TransactionUtils.java (hunk 4 above)
Comment (chaoxinhu) [reply src/main/java/com/webank/weid/util/TransactionUtils.java]: 其实没有。。这个是技术债，以后再改吧。。合约里面必须要定长。

File: src/main/java/com/webank/weid/util/TransactionUtils.java (hunk 3 above)
Comment (chaoxinhu) [reply src/main/java/com/webank/weid/util/TransactionUtils.java]: 这个记录一下issue

**Pull request merged**